pandas/indexes/base.py

```diff
@@ -1745,29 +1745,13 @@ def argsort(self, *args, **kwargs):
         return result.argsort(*args, **kwargs)
 
     def __add__(self, other):
-        if is_list_like(other):
-            warnings.warn("using '+' to provide set union with Indexes is "
-                          "deprecated, use '|' or .union()", FutureWarning,
-                          stacklevel=2)
-        if isinstance(other, Index):
```

Comment:
I would be a break in here (e.g. where it currently calls ``.union``) and see if we actually have test code that hits this. It obviously works, so prob ok to remove, just